### PR TITLE
Graph blas

### PR DESCRIPTION
**Pull request by swilly22**

### CODE REVIEW

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            if (col) *col = j ;
+            break ;
+        }
+        iter->p = 0 ;
+    }
+    
+    iter->col_idx = j ;
+    iter->nnz_idx++ ;
+
+    *depleted = false ;
```
Comment (jeffreylovitz) [inline deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: This line shouldn't be necessary, though I don't object to it existing.

File: src/Makefile [hunk 2]
```diff
@@ -96,7 +96,7 @@ $(LIBTRIEMAP):
 .PHONY: $(LIBTRIEMAP)
```
Comment (jeffreylovitz) [inline src/Makefile]: We should return to using the rm_malloc family, which I think would require the additional `-DCMAKE_C_FLAGS`:
`-DREDIS_MODULE_TARGET -DMALLOC=rm_malloc -DCALLOC=rm_calloc -DREALLOC=rm_realloc -DFREE=rm_free`

An additional include of `rmalloc.h` is also necessary, which we used to do in what would now be `deps/GraphBLAS/Source/GB.h:393`, but might also be viable as a `-I` CMAKE_C_FLAG.

File: deps/GraphBLAS/.gitignore [hunk 3]
```diff
@@ -1,69 +0,0 @@
-# Ignore these files:
```
Comment (jeffreylovitz) [inline deps/GraphBLAS/.gitignore]: Let's bring this file back, and add as much as we can to it. I think everything within `Source/Generated/` can be safely ignored, right?

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c (hunk 1 above)
Comment (swilly22) [reply deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: We can't assume anything about depleted value, suppose caller set depleted to true, and called 
GxB_MatrixTupleIter_next
we should update depleted.

**Pull request closed without merging**